sfbrigade/sfsbook
# adds login test (and changes a couple core files as necessary)

* The point of this commit was just to check that logging in results in a redirect from /login.html to /index.html.
* In order to do so, Search() in dba/password_index.go had to be generalized, which:
  * Fixes TODO in the code: "fix the layering violation of leaking bleve into the interface."
  * affects the code in server/login.go (it uses the general, non bleve, dba.PasswordSearchResult)
* We also had to implement mockPasswordIndex.Search() so the test can run.

Resolves #127

To test this test, run:

```
go get github.com/rjkroege/mocking 
cd server
go test -run TestServeHTTP
```

output should look like this:

```
2017/03/27 14:55:45 login.go:57: loginServer is handling request for /login.html
2017/03/27 14:55:45 login.go:63: handling Post of resource
2017/03/27 14:55:45 login.go:75: parsing form:
2017/03/27 14:55:45 login.go:77: 	 password 		 [password]
2017/03/27 14:55:45 login.go:77: 	 username 		 [test_user]
2017/03/27 14:55:45 login.go:119: username:  test_user has signed in
2017/03/27 14:55:45 login.go:146: usercookie &{28e5c7b9-9f39-4ebc-a9fb-5e7cfffa2687 175 2017-03-27 14:55:45.234172277 -0700 PDT test_user}
2017/03/27 14:55:45 login.go:163: login worked, redirecting to index
2017/03/27 14:55:45 templating.go:95: this is the template /login.html
PASS
ok  	github.com/sfbrigade/sfsbook/server	0.201s
```


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sfbrigade/sfsbook/128)
